lib/bambora/v1/payment_resource.rb

```diff
@@ -3,15 +3,32 @@
 module Bambora
   module V1
     class PaymentResource
+      CREDIT_CARD_ID_DEFAULT = 1
+
       # Summary: Create and modify payments.
       # Note: The link below links to all apis includding profiles and tokenization. There aren't great docs explaining
       #       the /payments endpoints alone.
       # Docs: https://dev.na.bambora.com/docs/references/payment_APIs/
       #       https://dev.na.bambora.com/docs/references/payment_SDKs/take_payments/?shell#
       # Endpoint: https://api.na.bambora.com/v1/payments
-      def initialize(client:, sub_path:)
+      def initialize(client:)
         @client = client
-        @sub_path = sub_path
+        @sub_path = '/v1/payments'
+      end
+
+      def make_payment(payment_data)
+        @client.request(method: :post, path: @sub_path, body: payment_data)
+      end
+
+      def make_payment_with_payment_profile(customer_code:, amount:)
+        make_payment(
+          amount: amount,
+          payment_method: 'payment_profile',
+          payment_profile: {
+            customer_code: customer_code,
+            card_id: CREDIT_CARD_ID_DEFAULT,
```

Comment:
Now using the `CREDIT_CARD_ID_DEFAULT`, rather than a magic number. Do you think I allow this to be passed in to the method as a parameter. My only issue with this, is that if we allow this, where do we draw the line in terms of what we do and don't allow to be passed in.

This method is meant to be more bespoke for our use case, however I do realize that given this is now OSS, it should be more generalized. If much more customization is needed, I would argue that using the lower level `#make_payment` method would be more useful, than having this method take on many many parameters. Unless we had a general options hash parameter, which could be merged in with the default data.

I'm okay to do something like `CREDIT_CARD_ID_DEFAULT || card_id`, with `card_id` being a `nil` default parameter in the method signature, if you think this would be the best way forward.

Reply:
@cassidycodes in this case, [according to the documentation](https://dev.na.bambora.com/docs/references/payment_APIs/v1-0-5/), when making a payment using a payment profile, there are 3 pieces of data that can be passed, of which 2 are required:

+ Customer Code (required)
+ Card ID (required)
+ Complete

So it does look like the Card ID is required. In our use case, we always keep it at `1`, because we currently do not allow multiple credit cards. I will make this a named argument with a default value of `1` like you said.

The Complete field indicates whether the payment should be completed or not. If it is not completed, the payment is made as a pre-authorization, meaning it will put a hold of the amount on the payment on the account, which will be subsequently released after a few days.

Since this is the only other field, I will also implement this as a named parameter. To prevent unintended charges, I will default this value to `false`.

Reply:
Ah, sorry, I didn't see this comment because of how GitHub links to changes from the notifications tab. Blargh. This all sounds good to me!

Reply:
Keep it specific to our use-case. Should we or others need to generalize this, additional PRs can be created.